### DISCUSSION TITLE
Display footer at bottom of page

### DISCUSSION THREAD

**Discussion by violetele** [thuliteio/doks#422] [Q&A] [answered]
Hi! I'm really appreciating Doks so far. One thing I'm stuck on trying to figure out is how to get the footer to display at the bottom of the page, even when the main page content is short.

I've gotten close by adding the `min-vh-100` Bootstrap class to the main content div (`div.wrap.container`), which gets close, but pushes the footer too far down. I got even closer by then accounting for the heights of the header and footer by using something like `min-height: calc(100vh - 60px - 64px)` on that div, but that's not great because it won't account for any layout changes, or for different devices. I'm pretty much at the limits of my Bootstrap knowledge here in dealing with the various flex layouts, so I figured I'd ask and see if you had any ideas.

Thanks!

**Comment by h-enk** [comment]
@violetele, does this help you: https://getbootstrap.com/docs/5.1/helpers/position/#fixed-bottom

**Reply by violetele** [reply]
It's close, but `fixed-bottom` puts the footer always at the bottom of the page, even when the content is longer than the viewport (so that the footer floats over long content). I'd like the footer to be fixed to the bottom of the page when the main page content is shorter than the viewport height, but also be at the bottom of the content whenever the main content is longer.

**Comment by violetele** [comment, marked as answer]
The solutions I've found suggest using Bootstrap's `d-flex` and `flex-column` classes on the body or a wrapper div, ~but those seem to break the layout of the `container` and `container-fluid` elements that Doks is using for the main content~.

This is an example of how other have solved this: https://radu.link/make-footer-stay-bottom-page-bootstrap/

_Actually_, I just tested this again, and it works on Firefox and on desktop Safari, but doesn't work on iPadOS Safari 🤦‍♀️  (the footer is displayed immediately below the bottom of the viewport).

**Reply by violetele** [reply]
Oh! Apparently iPadOS Safari doesn’t include the height of the expanded toolbar in the computed viewport size. So, the footer is displayed in the right place when the toolbar is collapsed, but is displayed below the bottom of the viewport when the toolbar is expanded. That’s kind of annoying, but it means that this solution using flex layouts does actually work (and is good enough, at least, on mobile Safari).

**Reply by h-enk** [reply]
Or, use https://getbootstrap.com/docs/5.1/utilities/flex/#align-items

**Comment by hitzhangjie** [comment]
This helps: https://radu.link/make-footer-stay-bottom-page-bootstrap/
